/*
 * Copyright 2016 The Kythe Authors. All rights reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package kythe.proto;

option go_package = "kythe.io/kythe/proto/buildinfo_go_proto";
option java_package = "com.google.devtools.kythe.proto";

// Details from the build tool used in a CompilationUnit.
// Its type is "kythe.io/proto/kythe.proto.BuildDetails".
message BuildDetails {
  // For build tools such as Bazel, that assign names to buildable targets, the
  // name of the build target that generated this compilation.
  string build_target = 1;

  // If known, the name or mnemonic of the generating rule.
  string rule_type = 2;

  // A pithy, user-meaningful description of the configuration or platform
  // this compilation is targeting, e.g. "android", "linux",
  // "mips-dec-ultrix4.2", etc.
  string build_config = 3;
}
